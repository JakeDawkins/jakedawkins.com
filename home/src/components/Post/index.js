import React, { PropTypes } from "react";
import ReactMarkdown from "react-markdown";
import { Link } from "react-router";
import "../../styles/_post.scss";

import Author from "../Author";
import data from "../../api/data.js"; //data for author

const Post = (props) =>
  <section className="post" itemScope itemType="http://schema.org/BlogPosting">
      <article role="article" id="scroll" className="post-content" itemProp="articleBody">
        { props.disableBackLink ? null :
          <p>
            <Link to={props.backLinkUrl ? props.backLinkUrl : "/posts"}>&lt;&nbsp;&nbsp;Back to Posts</Link>
          </p>
        }
        <ReactMarkdown source={props.markdown} />
        { props.disableBackLink ? null :
          <p style={{textAlign: "center"}}>
            <Link to="/posts">&lt;&nbsp;&nbsp;Back to Posts</Link>
          </p>
        }
      </article>

      {
        props.disableAuthor
          ? null
          : <Author
            userDisplayName={data.user.name}
            userImageUrl={data.user.image}
            userDescription={data.user.user_description}
            userEmail={data.user.email}
            />
      }
  </section>

//backLinkUrl not used yet
Post.propTypes = {
  markdown: PropTypes.string.isRequired,
  backLinkUrl: PropTypes.string,
  disableBackLink: PropTypes.bool,
  disableAuthor: PropTypes.bool,
}

export default Post;
