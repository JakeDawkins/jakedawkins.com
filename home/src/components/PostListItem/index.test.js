import Post from './';
import React from 'react';
import renderer from 'react-test-renderer';

const post = {
    postId: 0,
    publishedDateTime: "1475459431",
    publishedDay: "02",
    publishedMonth: "October",
    publishedYear: "2016",
    linkUrl: "/test", //relative link
    title: "tester",
    description: "very test much wow",
    readMoreText: "View on GitHub...", //title ex: "read more..."
    tags: ["web","react"],
    content: "madness",
  };

describe("Post", () => {

  it("should exist", () => {
    const post = renderer.create(
      <Post />
    );

    expect(post).toMatchSnapshot();
  });

  // XXX compare component with no props to this one soon
  it("should render props correctly", () => {

    const postWithProps = renderer.create(
      <Post {...post} />
    );

    expect(postWithProps).toMatchSnapshot();
  });

});
