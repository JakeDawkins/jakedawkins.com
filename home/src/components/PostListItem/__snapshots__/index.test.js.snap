exports[`Post should exist 1`] = `
<article
  className="post-item"
  itemProp="blogPost"
  itemScope="itemscope"
  itemType="http://schema.org/BlogPosting"
  role="article">
  <a
    className="datetime"
    onClick={[Function bound handleClick]}
    style={Object {}}>
    <meta
      content={undefined}
      itemProp="datePublished" />
    <time
      dateTime={undefined}>
      <span
        className="day">
         
         
      </span>
      <span
        className="month-year">
         
         
         
      </span>
    </time>
  </a>
  <div
    className="content">
    <a
      onClick={[Function bound handleClick]}
      style={Object {}}>
      <h2
        className="post-title"
        itemProp="name" />
    </a>
    <p
      className="description">
      undefined 
      <a
        className="link"
        onClick={[Function bound handleClick]}
        style={Object {}}>
        Read More...
      </a>
    </p>
    <div
      className="tags" />
  </div>
</article>
`;

exports[`Post should render props correctly 1`] = `
<article
  className="post-item"
  itemProp="blogPost"
  itemScope="itemscope"
  itemType="http://schema.org/BlogPosting"
  role="article">
  <a
    className="datetime"
    onClick={[Function bound handleClick]}
    style={Object {}}>
    <meta
      content="1475459431"
      itemProp="datePublished" />
    <time
      dateTime="1475459431">
      <span
        className="day">
         
        02
         
      </span>
      <span
        className="month-year">
         
        October
         
        2016
         
      </span>
    </time>
  </a>
  <div
    className="content">
    <a
      onClick={[Function bound handleClick]}
      style={Object {}}>
      <h2
        className="post-title"
        itemProp="name">
        tester
      </h2>
    </a>
    <p
      className="description">
      very test much wow 
      <a
        className="link"
        onClick={[Function bound handleClick]}
        style={Object {}}>
        View on GitHub...
      </a>
    </p>
    <div
      className="tags">
      <span>
        web
      </span>
      <span>
        react
      </span>
    </div>
  </div>
</article>
`;
